librz/bin/pdb/tpi.h

```diff
@@ -1,20 +1,695 @@
+	LF_ENUM = 0x1507, // type record for LF_ENUM
+	LF_DIMARRAY = 0x1508, // type record for dimensioned arrays
+	LF_PRECOMP = 0x1509, // type record describing inclusion of precompiled types
+	LF_ALIAS = 0x150A,
+	LF_DEFARG = 0x150B,
+	LF_FRIENDFCN = 0x150C, // subfield record for friend function
+	LF_MEMBER = 0x150D, // subfield record for non-static data members
+	LF_STMEMBER = 0x150E,
+	LF_METHOD = 0x150F, // subfield record for overloaded method list
+	LF_NESTTYPE = 0x1510, // type record for nested (scoped) type definition
+	LF_ONEMETHOD = 0x1511, // subfield record for nonoverloaded method
+	LF_NESTTYPEEX = 0x1512, // type record for nested (scoped) type definition, with attributes
+	LF_MEMBERMODIFY = 0x1513, // type record for modifications to members
+	LF_MANAGED = 0x1514,
+	LF_TYPESERVER2 = 0x1515, // type record describing using of a type server with v7 (GUID) signatures
+	LF_FUNCTION_ID = 0x1601,
+	LF_MEMBER_FUNCTION_ID = 0x1602,
+	LF_BUILD_INFO = 0x1603,
+	LF_SUBSTRING_LIST = 0x1604,
+	LF_STRING_ID = 0x1605,
+	LF_USER_DEFINED_TYPE_SOURCE_AND_LINE = 0x1606,
+	LF_USER_DEFINED_TYPE_MODULE_SOURCE_AND_LINE = 0x1607,
+	LF_CLASS_19 = 0x1608,
+	LF_STRUCTURE_19 = 0x1609,
+
+	/**     the following are numeric leaves.  They are used to indicate the
+	*      size of the following variable length data.  When the numeric
+	*      data is a single byte less than 0x8000, then the data is output
+	*      directly.  If the data is more the 0x8000 or is a negative value,
+	*      then the data is preceeded by the proper index.
+	*/
+	LF_CHAR = 0x8000, // signed character leaf
+	LF_SHORT = 0x8001, // signed short leaf
+	LF_USHORT = 0x8002, // unsigned short leaf
+	LF_LONG = 0x8003, // signed long leaf
+	LF_ULONG = 0x8004, // unsigned long leaf
+	LF_REAL32 = 0x8005, // real 32-bit leaf
+	LF_REAL64 = 0x8006, // real 64-bit leaf
+	LF_REAL80 = 0x8007, // real 80-bit leaf
+	LF_REAL128 = 0x8008, // real 128-bit leaf
+	LF_QUADWORD = 0x8009, // signed quad leaf
+	LF_UQUADWORD = 0x800A, // unsigned quad leaf
+	LF_REAL48 = 0x800B, // real 48-bit leaf
+	LF_COMPLEX32 = 0x800C, // complex 32-bit leaf
+	LF_COMPLEX64 = 0x800D, // complex 64-bit leaf
+	LF_COMPLEX80 = 0x800E, // complex 80-bit leaf
+	LF_COMPLEX128 = 0x800F, // complex 128-bit leaf
+	LF_VARSTRING = 0x8010, // variable length numeric field
+	LF_OCTWORD = 0x8017, // signed int128 leaf
+	LF_UOCTWORD = 0x8018, // unsigned int128 leaf
+	LF_DECIMAL = 0x8019,
+	LF_DATE = 0x801A,
+	LF_UTF8STRING = 0x801B,
+	LF_PAD0 = 0x00F0,
+	LF_PAD1 = 0x00F1,
+	LF_PAD2 = 0x00F2,
+	LF_PAD3 = 0x00F3,
+	LF_PAD4 = 0x00F4,
+	LF_PAD5 = 0x00F5,
+	LF_PAD6 = 0x00F6,
+	LF_PAD7 = 0x00F7,
+	LF_PAD8 = 0x00F8,
+	LF_PAD9 = 0x00F9,
+	LF_PAD10 = 0x00FA,
+	LF_PAD11 = 0x00FB,
+	LF_PAD12 = 0x00FC,
+	LF_PAD13 = 0x00FD,
+	LF_PAD14 = 0x00FE,
+	LF_PAD15 = 0x00FF,
+	LF_SIMPLE_TYPE = 0xEFFFFFFF, // Custom, hopefully it doesn't collide
+	LF_MAX = 0xFFFFFFFF
+} TpiLeafType;
+
+typedef union {
+	struct {
+#if __BYTE_ORDER == __LITTLE_ENDIAN
```

Comment:
I don't think this is a portable solution. Better not to use bitfields at all during the parsing.

Reply:
I have tested and this is worked on S390x.

Reply:
I meant the `__BYTE_ORDER` is not available everywhere, e.g. on MSVC. Thus, the Windows build will fail. I suggest reading this data without bitfields at all.